lib/improver/cube_combiner.py

```diff
@@ -47,72 +47,274 @@ class CubeCombiner(object):
 
     """
 
-    def __init__(self, operation):
+    def __init__(self, operation, warnings_on=False):
         """
         Create a CubeCombiner plugin
 
         Args:
             operation (str):
                 Operation (+, - etc) to apply to the incoming cubes.
+        Keyword Args:
+            warnings_on (bool):
+                If True output warnings for mismatching metadata.
 
         Raises:
             ValueError: Unknown operation.
+
         """
         possible_operations = ['+', 'add',
                                '-', 'subtract',
-                               '*', 'multiple',
+                               '*', 'multiply',
                                'max', 'min', 'mean']
         if operation in possible_operations:
             self.operation = operation
         else:
             msg = 'Unknown operation {}'.format(operation)
             raise ValueError(msg)
+        self.warnings_on = warnings_on
 
     def __repr__(self):
         """Represent the configured plugin instance as a string."""
-        desc = '<CubeCombiner: operation={}>'.format(self.operation)
+        desc = ('<CubeCombiner: operation=' +
+                '{}, warnings_on = {}>'.format(self.operation,
+                                               self.warnings_on))
         return desc
 
-    @staticmethod
-    def resolve_metadata_diff(cube1, cube2, revised_coords,
-                              revised_attributes,
-                              warnings_on=False):
-        """Resolve any differences in  metadata between cubes.
+    def resolve_metadata_diff(self, cube1, cube2):
```

Comment:
I'm happy to move these to cube_manipulation.py if we are certain that we would want to use these again. These use a dictionary supplied via a jsonfile to make changes to the Metadata in a cube because we want to supply this information via a CLI. I suspect that in other places we will just want to use iris directly to change the metadata rather than go through this more complicated interface.